### ISSUE TITLE
alternate download URL for "npm http GET https://registry.npmjs.org/grunt" etc.

### ISSUE THREAD

**Issue by mingmasplace** [eirslett/frontend-maven-plugin#301]
Hi, in our configuration the build machine doesn't have internet connection or proxy. Instead, it downloads from some internal web server. We successfully use <downloadRoot> to download node-v0.*.tar.gz and npm-*.tgz from the internal web server.

However, during npm install, it needs to access external URL such as "npm http GET https://registry.npmjs.org/grunt". Is there any way to config it to use alternate download URL for these requests?

**Comment by eirslett** [comment]
Try setting up something in `package.json`:

```javascript
{
  ...
  "publishConfig": {
    "registry": "http://my-registry-url"
  }
}
```

I'm not sure if that sets the npm registry you download packages from as well, and not just the repository you upload to. If that doesn't work, this should also be an option:

```xml
<arguments>install --registry=http://my-registry</arguments>
```

**Comment by mingmasplace** [comment]
Thanks! I was able to use the maven setting <arguments> to make some progress. Now it complains the certificate is untrusted. Here are the npm install command and the error messages. Under the same user account, "curl https://my-registry/node-and-npm/npm-install-registry/grunt-contrib-cssmin" works. Any ideas?

{noformat}
[INFO] Running 'npm install --ca="" --strict-ssl=true --registry=https://my-registry/resources/node-and-npm/npm-install-registry --color=false'

[INFO] npm http GET https://my-registry/node-and-npm/npm-install-registry/grunt-contrib-cssmin
[INFO] npm ERR! Error: CERT_UNTRUSTED
[INFO] npm ERR!     at SecurePair.<anonymous> (tls.js:1357:32)
{noformat}

**Comment by eirslett** [comment]
The problem with CA might be which certificate chains the node.js binary trusts? It might be that curl trusts it, but node doesn't.
In any case, these problems look more like they're general node/npm problems, not for this maven plugin in particular. So I'm sure you'll get more helpful answers if you ask ~~stackoverflow~~ somewhere else!

**Comment by Chandruhere4u** [comment]
I have got the exact issue now. Our organization dont have proxy and when I try "npm install" it is by default taking to registry.npmjs.org to which we dont have connection. is there a possibility to point to a internal repository and try from there.. I tried using "npm install --registry 'our internal repo' " but this didnt solve the problem. Any suggestion to solve this issue. I dont have internet connection from the server.
